src/Components/SmartComponents/SystemDetailsPage/SystemDetailsPage.js

```diff
@@ -31,25 +31,55 @@ class InventoryDetail extends React.Component {
         this.state = {
             inventoryId: this.props.match.params.inventoryId
         };
+
     }
 
     componentDidMount() {
         this.props.fetchSystemDetails(this.state.inventoryId);
     }
 
+    notifications = systemName => ({
+        exclude: {
+            success: {
+                variant: 'success',
+                title: this.props.intl.formatMessage(messages.notificationExcludeSuccess, { count: 1, systemName })
+            },
+            failure: {
+                variant: 'danger',
+                autoDismiss: false,
+                title: this.props.intl.formatMessage(messages.notificationExcludeFailureTitle, { count: 1, systemName }),
+                description: this.props.intl.formatMessage(messages.notificationDetailExcludeFailureBody, { count: 1 })
+            }
+        },
+        include: {
+            success: {
+                variant: 'success',
+                title: this.props.intl.formatMessage(messages.notificationIncludeSuccessTitle, { count: 1, systemName }),
+                description: this.props.intl.formatMessage(messages.notificationIncludeSuccessBody, { count: 1 })
+            },
+            failure: {
+                variant: 'danger',
+                autoDismiss: false,
+                title: this.props.intl.formatMessage(messages.notificationIncludeFailureTitle, { count: 1, systemName }),
+                description: this.props.intl.formatMessage(messages.notificationDetailIncludeFailureBody, { count: 1 })
+            }
+        }
+    });
+
     optOutSystem = value => {
         this.props.optOutSystem(this.state.inventoryId, value).then(() => {
+            const systemName = this.props.systemDetails.entity.display_name;
+
             this.props.fetchSystemDetails(this.state.inventoryId);
-            value === true
-                ? this.props.addNotification({
-                    variant: 'success',
-                    title: this.props.intl.formatMessage(messages.inventoryOptOutSuccess)
-                })
-                : this.props.addNotification({
-                    variant: 'success',
-                    title: this.props.intl.formatMessage(messages.inventoryResumeAnalysisTitle),
-                    description: this.props.intl.formatMessage(messages.inventoryResumeAnalysisDescription)
-                });
+            this.props.addNotification(value
```

Comment:
you created a helper to get the message based on the system state (included, excluded) which is great. you can make that helper a bit more helpful.

For example, now  you have to check it on every case

``` 
this.props.addNotification(value ? 
this.notifications(systemName).exclude.failure : this.notifications(systemName).include.failure); 
``` 
instead, you can create something like
`this.props.addNotification(getNotificationProps(value, systemName))
`
where `getNotificationProps` (or a better name)  will be responsible to decide which message to return based on your input